src/modules/navigator/mission_base.cpp

```diff
@@ -691,17 +693,20 @@ MissionBase::checkMissionRestart()
 void
 MissionBase::check_mission_valid()
 {
-	if (_navigator->get_mission_result()->mission_update_counter != _mission.mission_update_counter) {
-		MissionFeasibilityChecker missionFeasibilityChecker(_navigator, _dataman_client);
-
-		bool is_mission_valid =
-			missionFeasibilityChecker.checkMissionFeasible(_mission);
+	if ((_navigator->get_mission_result()->mission_update_counter != _mission.mission_update_counter)
+	    || (_navigator->get_mission_result()->geofence_update_counter != _mission.geofence_update_counter)
+	    || (_navigator->get_mission_result()->home_position_timestamp != _home_pos_sub.get().timestamp)) {
 
-		_navigator->get_mission_result()->valid = is_mission_valid;
 		_navigator->get_mission_result()->mission_update_counter = _mission.mission_update_counter;
+		_navigator->get_mission_result()->geofence_update_counter = _mission.geofence_update_counter;
+		_navigator->get_mission_result()->home_position_timestamp = _home_pos_sub.get().timestamp;
```

Comment:
You can use the existing home position:
```
_navigator->get_mission_result()->home_position_timestamp = _navigator->get_home_position()->timestamp;
```

Reply:
That means that we would tie this class again closer to the navigator module, although it might be decoupled in the future? Else i will apply your solution.

Reply:
I'm fine with it, but I also think this is easy to replace when the time comes to move it out.
I'm mostly concerned with having many `SubscriptionData` around, which can become expensive in terms of memory.

Reply:
@bkueng Yeah since memory can be a problem i fixed it to your solution